frontend/build.mjs

```diff
@@ -58,30 +58,54 @@ await buildInParallel(
             writeMetaFile: true,
             extraPlugins: [
                 {
-                    name: 'no-side-effects',
+                    name: 'denylist-imports',
                     setup(build) {
-                        // sideEffects in package.json lists files that _have_ side effects,
-                        // but we only want to mark lemon-ui as having no side effects,
-                        // so we'd have to list every other file and keep that up to date
-                        // no thanks!
-                        // a glob that negates the path doesn't seem to work
-                        // so based off a comment from the esbuild author here
-                        // https://github.com/evanw/esbuild/issues/1895#issuecomment-1003404929
-                        // we can add a plugin just for the toolbar build to mark lemon-ui as having no side effects
-                        // that will allow tree-shaking and reduce the toolbar bundle size
-                        // by over 40% at implementation time
-                        build.onResolve({ filter: /^(lib|@posthog)\/lemon-ui/ }, async (args) => {
-                            if (args.pluginData) {
-                                return
-                            } // Ignore this if we called ourselves
+                        // Explicit denylist of paths we don't want in the toolbar bundle
+                        const deniedPaths = [
+                            '~/lib/hooks/useUploadFiles',
+                            '~/queries/nodes/InsightViz/InsightViz',
+                            'lib/hog',
+                            'scenes/activity/explore/EventDetails',
+                        ]
 
-                            const { path, ...rest } = args
-                            rest.pluginData = true // Avoid infinite recursion
-                            const result = await build.resolve(path, rest)
+                        // Patterns to match for denying imports
+                        const deniedPatterns = [
+                            /monaco/,
+                            /scenes\/insights\/filters\/ActionFilter/,
+                            /lib\/components\/CodeSnippet/,
+                            /scenes\/session-recordings\/player/,
+                        ]
 
-                            result.sideEffects = false
+                        build.onResolve({ filter: /.*/ }, (args) => {
+                            const shouldDeny =
+                                deniedPaths.includes(args.path) ||
+                                deniedPatterns.some((pattern) => pattern.test(args.path))
 
-                            return result
+                            if (shouldDeny) {
+                                console.log(
```

Comment:
always comment on a file, because github doesn't let us thread at the top level @rafaeelaudibert 🤣 

> Does that cleanup happen locally too or only when building for prod? We'd want an easy way to see those are being removed, right?

yep, happens locally, the drawback is that if you break something by trying to use it then the toolbar often just dies

this, for sure, isn't an ideal solution. but I couldn't figure out a less ugly way of doing it

Reply:
in a perfect world we'd recreate the shape of the modules so that we could log that you were using it but then if you import a react component what should happen 🙈

Reply:
ok, figured out how to log when the toolbar imports something that will break it